ssl/t1_enc.c

```diff
@@ -82,6 +82,38 @@ static int tls1_generate_key_block(SSL *s, unsigned char *km, size_t num)
     return ret;
 }
 
+#ifndef OPENSSL_NO_KTLS
+ /* Count the number of records that were not processed yet from record boundary */
+static int count_unprocessed_records(SSL *s)
+{
+    SSL3_BUFFER *rbuf = RECORD_LAYER_get_rbuf(&s->rlayer);
+    unsigned char *packet = rbuf->buf + rbuf->offset;
+    int left = rbuf->left;
+    int count = 0;
+    int len;
+
+    if (rbuf == NULL || left == 0)
+        return count;
+
+    while (left > 0) {
+        /* This should never happen */
+        if (left < 5)
+            return -1;
+
+        len = ((unsigned int)(*(packet + 3))) << 8;
+        len |= *(packet + 4);
```

Comment:
I think all of this should be converted to use the `PACKET` API instead.

Reply:
A huge benefit of the PACKET API is that it makes the code immensely easier to review and to maintain.  I will happily take long, easy-to-review code over short-but-opaque code every day.

Reply:
Done.